certificate.tf

```
@@ -0,0 +1,46 @@
+#-------------------------------------------------------------------------------
+# Create the ACM certificate used by the FreeIPA cluster.
+#-------------------------------------------------------------------------------
+
+# AWS certificate for load balancers
+resource "aws_acm_certificate" "ipa" {
+  provider = aws.sharedservicesprovisionaccount
+
+  domain_name = "ipa.${var.cool_domain}"
+  subject_alternative_names = [
+    "ipa-ca.${var.cool_domain}",
+    "ipa0.${var.cool_domain}",
+    "ipa1.${var.cool_domain}",
+    "ipa2.${var.cool_domain}",
+  ]
+  validation_method = "DNS"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_route53_record" "ipa_cert_validation" {
+  for_each = {
+    for dvo in aws_acm_certificate.ipa.domain_validation_options : dvo.domain_name => {
+      name   = dvo.resource_record_name
+      record = dvo.resource_record_value
+      type   = dvo.resource_record_type
+    }
+  }
+  provider = aws.public_dns
+
+  allow_overwrite = true
+  name            = each.value.name
+  records         = [each.value.record]
+  ttl             = 60
```

Should we be using `var.ttl` here?
```suggestion
  ttl             = var.ttl
```

---

We should not.  These are just records that are put in place so that ACM can verify we own the domain.  If `var.ttl` is set to a large value because we don't expect the FreeIPA-specific DNS records to change, we would still want these records to have a low TTL so that redeployments can happen in a timely manner.

I can add a comment explaining this.

---

Got it- thanks, a comment will be helpful there.

---

Please see commit ea461b50a2abe01cedaf12aa5524f22000007220.